src/transformers/models/auto/auto_factory.py

```diff
@@ -520,15 +522,20 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
             if kwargs.get("quantization_config", None) is not None:
                 _ = kwargs.pop("quantization_config")
 
-            config, kwargs = AutoConfig.from_pretrained(
-                pretrained_model_name_or_path,
-                return_unused_kwargs=True,
-                trust_remote_code=trust_remote_code,
-                code_revision=code_revision,
-                _commit_hash=commit_hash,
-                **hub_kwargs,
-                **kwargs,
-            )
+            if from_gguf is None:
+                config, kwargs = AutoConfig.from_pretrained(
+                    pretrained_model_name_or_path,
+                    return_unused_kwargs=True,
+                    trust_remote_code=trust_remote_code,
+                    code_revision=code_revision,
+                    _commit_hash=commit_hash,
+                    **hub_kwargs,
+                    **kwargs,
+                )
+            else:
+                gguf_path = cached_file(pretrained_model_name_or_path, from_gguf, **hub_kwargs)
+
+                config = load_and_convert_gguf_config(gguf_path)
```

Comment:
I wonder if these changes wouldn't be cleaner if they were directly in the `AutoConfig` definition rather than in the auto factory

Reply:
Makes sense ! Done !